### PR TITLE
misc updates to OSPP profile

### PR DESCRIPTION
**Pull request by shawndwells**
- edit -

To unblock ongoing development, updated PR to merge in the package checks.

Add audit rules for successful syscalls

XCCDF for:

- [x] openat O_CREAT https://github.com/ComplianceAsCode/content/issues/4557
- [x]  open_by_handle_at O_CREAT https://github.com/ComplianceAsCode/content/issues/4558
- [x] open O_CREAT https://github.com/ComplianceAsCode/content/issues/4559
- [x] openat O_TRUNC_WRITE https://github.com/ComplianceAsCode/content/issues/4561
- [x] open_by_handle_at O_TRUNC_WRITE https://github.com/ComplianceAsCode/content/issues/4562
- [x] open O_TRUNC_WRITE https://github.com/ComplianceAsCode/content/issues/4560
- [x] open https://github.com/ComplianceAsCode/content/issues/4549
- [x] creat https://github.com/ComplianceAsCode/content/issues/4552
- [x] truncate https://github.com/ComplianceAsCode/content/issues/4553
- [x] ftruncate https://github.com/ComplianceAsCode/content/issues/4554
- [x] openat https://github.com/ComplianceAsCode/content/issues/4555
- [x] open_by_handle_at https://github.com/ComplianceAsCode/content/issues/4556
- [x] unlink https://github.com/ComplianceAsCode/content/issues/4549
- [x] unlinkat https://github.com/ComplianceAsCode/content/issues/4548
- [x] rename https://github.com/ComplianceAsCode/content/issues/4550
- [x] renameat https://github.com/ComplianceAsCode/content/issues/4551
- [x] chmod https://github.com/ComplianceAsCode/content/issues/4533
- [x] fchmod https://github.com/ComplianceAsCode/content/issues/4534
- [x] fchmodat https://github.com/ComplianceAsCode/content/issues/4535
- [x] setxattr https://github.com/ComplianceAsCode/content/issues/4536
- [x] lsetxattr https://github.com/ComplianceAsCode/content/issues/4537
- [x] fsetxattr https://github.com/ComplianceAsCode/content/issues/4538
- [x] removexattr https://github.com/ComplianceAsCode/content/issues/4538
- [x] lremovexattr https://github.com/ComplianceAsCode/content/issues/4539
- [x] fremotexattr https://github.com/ComplianceAsCode/content/issues/4540
- [x] lchown https://github.com/ComplianceAsCode/content/issues/4519
- [x] fchown https://github.com/ComplianceAsCode/content/issues/4521
- [x] chown https://github.com/ComplianceAsCode/content/issues/4522
- [x] fchownat https://github.com/ComplianceAsCode/content/issues/4523

Misc
- Resolves nodev on /var via https://github.com/ComplianceAsCode/content/issues/4489
- Resolves perf_event_paranoid https://github.com/ComplianceAsCode/content/issues/4491
- resolves JIT via https://github.com/ComplianceAsCode/content/issues/4494
- audit use of seunshare via https://github.com/ComplianceAsCode/content/issues/4513

### CODE REVIEW

**Comment by shawndwells** [comment]
Taking a break for tonight. Should be merge-able pending jenkins, so removing the ``WIP`` flag in case anyone can review.

**Comment by jan-cerny** [comment]
the test scenarios need to be moved as well

**Comment by shawndwells** [comment]
> the test scenarios need to be moved as well

Done. 

Wonder if the test cases could be moved inside the rule directory. Didn't even know they existed!

**Comment by jan-cerny** [comment]
> Wonder if the test cases could be moved inside the rule directory. Didn't even know they existed!

I agree. We already have this idea, I hope it will get done soon, because it will be also easier to use the scenarios during the  PR reviews if they're in the same place as the rule.

**Comment by shawndwells** [comment]
> It is really hard to follow what is happening in [7b47a6b](https://github.com/ComplianceAsCode/content/commit/7b47a6bbb2b385011a46b53017b7467e166b2720).
> Commit message says OSPP updates, but a bunch of files are changing directory.
> 
> I understood that they are being grouped under `linux_os/guide/system/auditing/auditd_configure_rules/audit_file_modification/` but some files even change name, like
> 
> ```
> linux_os/guide/system/auditing/auditd_configure_rules/audit_unsuccessful_file_modification/audit_rules_unsuccessful_file_modification_openat/bash/shared.sh → linux_os/guide/system/auditing/auditd_configure_rules/audit_file_modification/audit_rules_unsuccessful_file_modification_open_by_handle_at/bash/shared.sh
> ```

Odd. Shows that it was only a directory rename in the PR:

image

Looking into this.

File: applications/openshift/ocp-permissions/ocp-files/file_owner_master_api_server/rule.yml [hunk 1]
```diff
@@ -14,7 +14,7 @@ rationale: |-
 severity: medium
 
 identifiers:
-    cce@ocp3: 80577-0
```
[the remark this answers is not in the document]
Comment (shawndwells) [reply applications/openshift/ocp-permissions/ocp-files/file_owner_master_api_server/rule.yml]: Duplicate as identified in https://github.com/ComplianceAsCode/content/pull/4577

File: applications/openstack/keystone_container/container_keystone_lockout_duration/rule.yml [hunk 2]
```diff
@@ -23,7 +23,7 @@ rationale: |-
 severity: medium
 
 identifiers:
-    cce@osp13: 80995-4
```
[the remark this answers is not in the document]
Comment (shawndwells) [reply applications/openstack/keystone_container/container_keystone_lockout_duration/rule.yml]: Duplicate as identified in https://github.com/ComplianceAsCode/content/pull/4577

File: firefox/guide/firefox/firefox_preferences-dod_root_certificate/firefox_preferences-enable_ca_trust/rule.yml [hunk 3]
```diff
@@ -17,7 +17,7 @@ rationale: |-
 severity: medium
 
 identifiers:
-    cce: 27457-1
```
[the remark this answers is not in the document]
Comment (shawndwells) [reply firefox/guide/firefox/firefox_preferences-dod_root_certificate/firefox_preferences-enable_ca_trust/rule.yml]: Another from https://github.com/ComplianceAsCode/content/pull/4577

**Comment by yuumasato** [comment]
Most files are renamed correctly. Just a few are strange:
From git log it may be easier to spot, these are the ones I found:
```
rename from linux_os/guide/system/auditing/auditd_configure_rules/audit_unsuccessful_file_modification/audit_rules_unsuccessful_file_modification_open_by_handle_at_o_creat/bash/shared.sh
rename to linux_os/guide/system/auditing/auditd_configure_rules/audit_file_modification/audit_rules_successful_file_modification_openat/bash/shared.sh

rename from linux_os/guide/system/auditing/auditd_configure_rules/audit_unsuccessful_file_modification/audit_rules_unsuccessful_file_modification_open_by_handle_at_o_trunc_write/bash/shared.sh
rename to linux_os/guide/system/auditing/auditd_configure_rules/audit_file_modification/audit_rules_successful_file_modification_truncate/bash/shared.sh

rename from linux_os/guide/system/auditing/auditd_configure_rules/audit_unsuccessful_file_modification/audit_rules_unsuccessful_file_modification_open_by_handle_at_rule_order/bash/shared.sh
rename to linux_os/guide/system/auditing/auditd_configure_rules/audit_file_modification/audit_rules_successful_file_modification_unlink/bash/shared.sh


```

**Comment by shawndwells** [comment]
> Most files are renamed correctly. Just a few are strange:
> From git log it may be easier to spot, these are the ones I found:
> 
> ```
> rename from linux_os/guide/system/auditing/auditd_configure_rules/audit_unsuccessful_file_modification/audit_rules_unsuccessful_file_modification_open_by_handle_at_o_creat/bash/shared.sh
> rename to linux_os/guide/system/auditing/auditd_configure_rules/audit_file_modification/audit_rules_successful_file_modification_openat/bash/shared.sh
> 
``linux_os/guide/system/auditing/auditd_configure_rules/audit_file_modification/audit_rules_successful_file_modification_openat/bash/shared.sh`` doesn't exist in this PR....

> rename from linux_os/guide/system/auditing/auditd_configure_rules/audit_unsuccessful_file_modification/audit_rules_unsuccessful_file_modification_open_by_handle_at_o_trunc_write/bash/shared.sh
> rename to linux_os/guide/system/auditing/auditd_configure_rules/audit_file_modification/audit_rules_successful_file_modification_truncate/bash/shared.sh
> 
Neither does this

> rename from linux_os/guide/system/auditing/auditd_configure_rules/audit_unsuccessful_file_modification/audit_rules_unsuccessful_file_modification_open_by_handle_at_rule_order/bash/shared.sh
> rename to linux_os/guide/system/auditing/auditd_configure_rules/audit_file_modification/audit_rules_successful_file_modification_unlink/bash/shared.sh
> ```

Or this

**Comment by shawndwells** [comment]
> > Most files are renamed correctly. Just a few are strange:
> > From git log it may be easier to spot, these are the ones I found:
> > ```
> > rename from linux_os/guide/system/auditing/auditd_configure_rules/audit_unsuccessful_file_modification/audit_rules_unsuccessful_file_modification_open_by_handle_at_o_creat/bash/shared.sh
> > rename to linux_os/guide/system/auditing/auditd_configure_rules/audit_file_modification/audit_rules_successful_file_modification_openat/bash/shared.sh
> > ```
> 
> `linux_os/guide/system/auditing/auditd_configure_rules/audit_file_modification/audit_rules_successful_file_modification_openat/bash/shared.sh` doesn't exist in this PR....
> 

Seems find in the PR.... renamed to ``.//linux_os/guide/system/auditing/auditd_configure_rules/audit_file_modification/audit_rules_unsuccessful_file_modification_open_by_handle_at_o_trunc_write/bash/shared.sh``....

> > rename from linux_os/guide/system/auditing/auditd_configure_rules/audit_unsuccessful_file_modification/audit_rules_unsuccessful_file_modification_open_by_handle_at_o_trunc_write/bash/shared.sh
> > rename to linux_os/guide/system/auditing/auditd_configure_rules/audit_file_modification/audit_rules_successful_file_modification_truncate/bash/shared.sh
> 
> Neither does this
> 
> > rename from linux_os/guide/system/auditing/auditd_configure_rules/audit_unsuccessful_file_modification/audit_rules_unsuccessful_file_modification_open_by_handle_at_rule_order/bash/shared.sh
> > rename to linux_os/guide/system/auditing/auditd_configure_rules/audit_file_modification/audit_rules_successful_file_modification_unlink/bash/shared.sh
> 
> Or this

**Comment by yuumasato** [comment]
> > Most files are renamed correctly. Just a few are strange:
> > From git log it may be easier to spot, these are the ones I found:
> > ```
> > rename from linux_os/guide/system/auditing/auditd_configure_rules/audit_unsuccessful_file_modification/audit_rules_unsuccessful_file_modification_open_by_handle_at_o_creat/bash/shared.sh
> > rename to linux_os/guide/system/auditing/auditd_configure_rules/audit_file_modification/audit_rules_successful_file_modification_openat/bash/shared.sh
> > ```
> 
> `linux_os/guide/system/auditing/auditd_configure_rules/audit_file_modification/audit_rules_successful_file_modification_openat/bash/shared.sh` doesn't exist in this PR....
> 
> > rename from linux_os/guide/system/auditing/auditd_configure_rules/audit_unsuccessful_file_modification/audit_rules_unsuccessful_file_modification_open_by_handle_at_o_trunc_write/bash/shared.sh
> > rename to linux_os/guide/system/auditing/auditd_configure_rules/audit_file_modification/audit_rules_successful_file_modification_truncate/bash/shared.sh
> 
> Neither does this
> 
> > rename from linux_os/guide/system/auditing/auditd_configure_rules/audit_unsuccessful_file_modification/audit_rules_unsuccessful_file_modification_open_by_handle_at_rule_order/bash/shared.sh
> > rename to linux_os/guide/system/auditing/auditd_configure_rules/audit_file_modification/audit_rules_successful_file_modification_unlink/bash/shared.sh
> 
> Or this

Yeah, I see now that you remove them in 88f837c97fb48697b4dd22226dcd84be406ae3ce.


> .... all of these seem fine?

I have compared the DataStreams (with `utils/compare_generated.sh`) before and after this PR, and nothing is removed, just added.
Changes are ok, as far as I can tell.

So yeah, it seems they are all there.

I'll squash commits when merging this PR, to remove confusion on renamed and deleted files in 7b47a6bbb2b385011a46b53017b7467e166b2720 and 88f837c97fb48697b4dd22226dcd84be406ae3ce.

**Comment by yuumasato** [comment]
@shawndwells Please, try to make the PRs more self contained, and separated in reasonable commits.
